bug: test test_optimized_model fails since HF transformer 4.26

### Description

test test_optimized_model fails since HF transformer 4.26 (released a few days ago)
Test done on current and today pytorch nightly version (both crashing).


### Steps to reproduce

```shell
❯ pytest test/test_model_optimization.py
```

### Expected Behavior

no crash

### Actual Behavior

```log
============================= test session starts ==============================
platform linux -- Python 3.9.16, pytest-7.2.1, pluggy-1.0.0
rootdir: /mnt/workspace/kernl
plugins: anyio-3.6.2
collected 1 item                                                               

test/test_model_optimization.py F                                                                                                                                               [100%]

====================================================================================== FAILURES =======================================================================================
________________________________________________________________________________ test_optimized_model _________________________________________________________________________________

    @set_seed()
    def test_optimized_model():
        shape = (1, 128)
        model_name = "bert-base-uncased"
        model = AutoModel.from_pretrained(model_name).eval().cuda()
        optimized_model = AutoModel.from_pretrained(model_name).eval().cuda()
        optimize_model(optimized_model)
    
        with torch.inference_mode(), torch.cuda.amp.autocast(enabled=True, dtype=torch.float16, cache_enabled=True):
            inputs = {
                "input_ids": torch.randint(2, 10000, shape, device="cuda", dtype=torch.long),
                "attention_mask": torch.ones(shape, device="cuda", dtype=torch.long),
            }
            output_pytorch = model(**inputs)
>           output_optimized = optimized_model(**inputs)

test/test_model_optimization.py:39: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/home/geantvert/.local/share/virtualenvs/kernl/lib/python3.9/site-packages/torch/nn/modules/module.py:1488: in _call_impl
    return forward_call(*args, **kwargs)
/home/geantvert/.local/share/virtualenvs/kernl/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:211: in _fn
    return fn(*args, **kwargs)
src/kernl/model_optimization.py:64: in run
    return model.forward_original(*args, **kwargs)
/home/geantvert/.local/share/virtualenvs/kernl/lib/python3.9/site-packages/transformers/models/bert/modeling_bert.py:911: in forward
    @add_start_docstrings_to_model_forward(BERT_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (), kwargs = {}, prior = <function convert_frame.<locals>._convert_frame at 0x7fbf5866a820>, backend_ctx = <contextlib.nullcontext object at 0x7fbf5866f940>
dynamic_ctx = <contextlib._GeneratorContextManager object at 0x7fbf583c7520>

    @functools.wraps(fn)
    def _fn(*args, **kwargs):
        if (
            not isinstance(self, DisableContext)
            and torch.fx._symbolic_trace.is_fx_tracing()
        ):
            if config.error_on_nested_fx_trace:
                raise RuntimeError(
                    "Detected that you are using FX to symbolically trace "
                    "a dynamo-optimized function. This is not supported at the moment."
                )
            else:
                return fn(*args, **kwargs)
    
        on_enter()
        prior = set_eval_frame(callback)
        backend_ctx = backend_ctx_ctor()
        backend_ctx.__enter__()
        dynamic_ctx = enable_dynamic(self.dynamic)
        dynamic_ctx.__enter__()
        try:
>           return fn(*args, **kwargs)
E           TypeError: <lambda>() missing 1 required positional argument: 'args'

/home/geantvert/.local/share/virtualenvs/kernl/lib/python3.9/site-packages/torch/_dynamo/eval_frame.py:211: TypeError
-------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertModel: ['cls.predictions.transform.LayerNorm.bias', 'cls.seq_relationship.bias', 'cls.predictions.transform.dense.weight', 'cls.seq_relationship.weight', 'cls.predictions.decoder.weight', 'cls.predictions.bias', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight']
- This IS expected if you are initializing BertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertModel: ['cls.predictions.transform.LayerNorm.bias', 'cls.seq_relationship.bias', 'cls.predictions.transform.dense.weight', 'cls.seq_relationship.weight', 'cls.predictions.decoder.weight', 'cls.predictions.bias', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight']
- This IS expected if you are initializing BertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
================================================================================== warnings summary ===================================================================================
test/test_model_optimization.py::test_optimized_model
  /home/geantvert/.local/share/virtualenvs/kernl/lib/python3.9/site-packages/torch/cuda/graphs.py:82: UserWarning: The CUDA Graph is empty. This ususally means that the graph was attempted to be captured on wrong device or stream. (Triggered internally at ../aten/src/ATen/cuda/CUDAGraph.cpp:191.)
    super(CUDAGraph, self).capture_end()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================== short test summary info ===============================================================================
FAILED test/test_model_optimization.py::test_optimized_model - TypeError: <lambda>() missing 1 required positional argument: 'args'
============================================================================ 1 failed, 1 warning in 5.67s =============================================================================
```

### Your environment

- Operating system and version (e.g. Ubuntu 20.04.2 LTS):
- Python version (e.g. Python 3.8.10):
- Python package manager (e.g. pip 21.1.2):
- etc.


### Self-service

- [X] I would be willing to help fix this bug myself.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM nvidia/cuda:11.6.1-devel-ubuntu20.04
+FROM nvidia/cuda:12.0.0-devel-ubuntu22.04
 ENV DEBIAN_FRONTEND=noninteractive
 
 ENV CUDA_INSTALL_PATH=/usr/local/cuda/
@@ -21,10 +21,10 @@ RUN apt-get install -y git \
     python3.9-dev \
     nano
 
-RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 && \
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.10 1 && \
   update-alternatives --install /usr/bin/python python /usr/bin/python3.9 2 && \
   update-alternatives --set python /usr/bin/python3.9 && \
-  update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1 && \
+  update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.10 1 && \
   update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 2 && \
   update-alternatives --set python3 /usr/bin/python3.9
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -30,7 +30,7 @@ RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 &&
 
 RUN python3.9 -m ensurepip --default-pip --upgrade
 
-RUN pip install --pre torch==2.0.0.dev20230119+cu117 --extra-index-url https://download.pytorch.org/whl/nightly/cu117
+RUN pip install --pre torch==2.0.0.dev20230128+cu117 --extra-index-url https://download.pytorch.org/whl/nightly/cu117
 
 
 WORKDIR /syncback

--- a/README.md
+++ b/README.md
@@ -35,7 +35,6 @@ If you prefer `Docker`:
 
 ```shell
 # build
-make docker_build
 DOCKER_BUILDKIT=1 docker build -t kernl .
 # run
 docker run --rm -it --gpus all -v $(pwd):/kernl kernl

--- a/experimental/whisper/README.md
+++ b/experimental/whisper/README.md
@@ -7,6 +7,6 @@ To run the notebook through shell, use the following command:
 DOCKER_BUILDKIT=1 docker build -t kernl .
 docker run --rm -it --gpus all -v $(pwd):/kernl kernl
 apt install libsndfile1-dev # used by a Python audio dependency
-pip install datasets soundfile librosa -q
+pip install datasets soundfile librosa jupyter notebook
 jupyter nbconvert --execute --clear-output experimental/whisper/speedup.ipynb --log-level=10
 ```

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 triton==2.0.0.dev20221202
-torch==2.0.0.dev20230119+cu117
+torch==2.0.0.dev20230128+cu117
 pytest
 tabulate
 termcolor

--- a/src/kernl/optimizer/cuda_graph.py
+++ b/src/kernl/optimizer/cuda_graph.py
@@ -72,7 +72,7 @@ def cuda_graphs_wrapper(model: Callable, inputs: Union[list[torch.Tensor], tuple
         f = cudagraphify_impl(
             model=lambda args: model(*args), inputs=inputs, static_input_idxs=tuple(range(len(inputs)))
         )
-        return lambda args: f(get_static_inputs(args))
+        return lambda *args: f(get_static_inputs(args))
 
     compiled_fn = None
 

--- a/test/test_attention.py
+++ b/test/test_attention.py
@@ -206,7 +206,7 @@ def test_benchmark_skinny_cross_attention(benchmark, implementation, shape):
     output = torch.empty_like(q)
     fn = implementations_skinny_cross_attention[implementation](output, sm_scale)
     r = cuda_graphs_wrapper(fn, [q, k, v])
-    _ = r([q, k, v])[0]
-    result = benchmark(r, [q, k, v])[0]
+    _ = r(q, k, v)[0]
+    result = benchmark(r, q, k, v)[0]
 
     assert_all_close(a=expected, b=result.float(), atol=1e-2)

--- a/test/test_layer_norm.py
+++ b/test/test_layer_norm.py
@@ -67,7 +67,7 @@ def test_benchmark_layer_norm(benchmark, shape: int, dtype, cuda_graphs: bool, i
     if cuda_graphs:
         run = cuda_graphs_wrapper(model=fn, inputs=[x])
         # CUDA graphs wraps output in a tuple
-        fn = lambda tensor: run([tensor])[0]  # noqa: E731
+        fn = lambda tensor: run(tensor)[0]  # noqa: E731
 
     value = benchmark(fn, x)
     assert_all_close(value.float(), expected, atol=1e-1)
@@ -101,7 +101,7 @@ def test_benchmark_rms_norm(benchmark, shape: int, dtype, cuda_graphs: bool, imp
     if cuda_graphs:
         run = cuda_graphs_wrapper(model=fn, inputs=[x])
         # CUDA graphs wraps output in a tuple
-        fn = lambda tensor: run([tensor])[0]  # noqa: E731
+        fn = lambda tensor: run(tensor)[0]  # noqa: E731
 
     value = benchmark(fn, x)
     assert_all_close(value.float(), expected, atol=1e-1)

--- a/test/test_linear_layer.py
+++ b/test/test_linear_layer.py
@@ -92,7 +92,7 @@ def test_benchmark(
     if cuda_graphs:
         run = cuda_graphs_wrapper(model=fn, inputs=[x])
         # CUDA graphs wraps output in a tuple
-        fn = lambda tensor: run([tensor])[0]  # noqa: E731
+        fn = lambda tensor: run(tensor)[0]  # noqa: E731
 
     value = benchmark(fn, x)
 

